### PR TITLE
Disallow default OCI repo translation to docker.io.

### PR DESCRIPTION
**Pull request by hpidcock**
Matching on empty domain portion of OCI image paths has been problematic in the past. This change requires an explicit domain part of the OCI image path to allow for proper matching.

## QA steps

Test bootstrap and upgrade with a real docker.io repo.

## Documentation changes

Changes to the caas-image-repo documentation possibly required.

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2037744

**Jira card:** JUJU-

### CODE REVIEW

File: docker/registry/internal/acr.go [hunk 1]
```diff
@@ -18,15 +18,19 @@ type azureContainerRegistry struct {
...
+func newAzureContainerRegistry(repoDetails docker.ImageRepoDetails, transport http.RoundTripper) (RegistryInternal, error) {
+	c, err := newBase(repoDetails, transport, normalizeRepoDetailsAzure)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &azureContainerRegistry{c}, nil
 }
 
-func normalizeRepoDetailsAzure(repoDetails *docker.ImageRepoDetails) {
+func normalizeRepoDetailsAzure(repoDetails *docker.ImageRepoDetails) error {
```
Comment (tlm) [inline docker/registry/internal/acr.go:29]: Don't understand the new error return here?

Comment (tlm) [reply docker/registry/internal/acr.go:29]: Never mind I understand now.

**Review by tlm — APPROVED** [review]
lgtm

**Comment by tlm** [comment]
This seems good to me and tested. I think we should get some docs generated and possibly a discourse post.

**Comment by hpidcock** [comment]
> caas-image-repo

I'll add a jira card for it. Thanks.

**Pull request merged**